spec/unit/MatrixAction.spec.js

```diff
@@ -87,10 +88,40 @@ describe("MatrixAction", function() {
         action.formatMentions({
             "JCDenton": "@jc.denton:unatco.gov"
         });
-        expect(action.text).toEqual("JCDenton, meet JCDenton");
+        expect(action.text).toEqual("jc.denton, meet jc.denton");
         expect(action.htmlText).toEqual(
-            "<a href=\"https://matrix.to/#/@jc.denton:unatco.gov\">JCDenton</a>," +
-            " meet <a href=\"https://matrix.to/#/@jc.denton:unatco.gov\">JCDenton</a>"
+            "<a href=\"https://matrix.to/#/@jc.denton:unatco.gov\">jc.denton</a>," +
+            " meet <a href=\"https://matrix.to/#/@jc.denton:unatco.gov\">jc.denton</a>"
+        );
+    });
+    it("should not highlight mentions in a URL with www.", () => {
+        let action = new MatrixAction(
+            "message",
+            "Go to http://www.JCDenton.com",
+            "Go to <a href='http://www.JCDenton.com'>my website</a>",
+            null
+        );
+        action.formatMentions({
+            "JCDenton": "@jc.denton:unatco.gov"
+        });
+        expect(action.text).toEqual("Go to http://www.JCDenton.com");
+        expect(action.htmlText).toEqual(
+            "Go to <a href='http://www.JCDenton.com'>my website</a>"
+        );
+    });
+    it("should not highlight mentions in a URL with http://", () => {
+        let action = new MatrixAction(
+            "message",
+            "Go to http://JCDenton.com",
+            "Go to <a href='http://JCDenton.com'>my website</a>",
+            null
+        );
+        action.formatMentions({
+            "JCDenton": "@jc.denton:unatco.gov"
+        });
+        expect(action.text).toEqual("Go to http://JCDenton.com");
+        expect(action.htmlText).toEqual(
+            "Go to <a href='http://JCDenton.com'>my website</a>"
         );
```

Comment:
a test for making sure it also actually uses the profile where possible would be nice